### PR TITLE
EMR Lambda and Terraform CD

### PR DESCRIPTION
**Pull request by eganjs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,8 +22,31 @@ CircleCI requires an AWS IAM User to run the terraform so create an IAM User cal
...
+```
+export TF_VAR_PASSWORD=<ais_to_heatmaps_parameters_password>
+
+./gradlew deploy # :deployment:terraformPlan :deployment:terraformApply
+```
+
+To decrypt `parameters.json.enc`:
+
+```
+./decrypt_parameters.sh # Supplying ais_to_heatmaps_parameters_password when prompted
```
Comment (ciaranevans) [inline README.md]: # Comment not needed now

File: README.md [hunk 2]
```diff
@@ -22,8 +22,31 @@ CircleCI requires an AWS IAM User to run the terraform so create an IAM User cal
...
+```
+
+
+To encrypt `parameters.secret.json`:
+
+```
+./encrypt_parameters.sh # Supplying ais_to_heatmaps_parameters_password when prompted
```
Comment (ciaranevans) [inline README.md]: Ditto

File: buildSrc/src/main/kotlin/TerraformPlugin.kt [hunk 3]
```diff
@@ -1,53 +1,168 @@
...
+}
+
+open class TerraformApplyTask @Inject constructor() : TerraformTask() {
+    init {
+        this.dependsOn(terraformTaskInit)
+    }
+
+    @Option(option = "auto-approve", description = "Skip interactive approval of plan before applying. Overrides task configuration.")
+    @get:Input
+    var autoApprove: Boolean = false
```
Comment (JoelClemence) [inline buildSrc/src/main/kotlin/TerraformPlugin.kt]: rename ci? Better describes intentions

Comment (eganjs) [reply buildSrc/src/main/kotlin/TerraformPlugin.kt]: I see where you're coming from but that would abstract away what it means to use that flag

Comment (JoelClemence) [reply buildSrc/src/main/kotlin/TerraformPlugin.kt]: Maybe document the intentions in Readme then?

Comment (eganjs) [reply buildSrc/src/main/kotlin/TerraformPlugin.kt]: Yep, that sounds good.

File: README.md [hunk 4]
```diff
@@ -32,19 +32,36 @@ export TF_VAR_PASSWORD=<ais_to_heatmaps_parameters_password>
...
+
+#### Terraform Gradle tasks:
+
+**:deployment:terraformClean** - Removes all terraform binaries
+
+**:deployment:terraformInit** - initialises Terraform, *--interactive* prints out the output of the initialisation to Standard out
+
+**:deployment:terraformValidate** - validates terraform templates
+
+**:deployment:terraformDownload** - downloads Terraform so that the plugin can us it
```
Comment (ciaranevans) [inline README.md]: use*

**Pull request merged**